src/main/java/messenger/MessengerDatabase.java

```diff
@@ -0,0 +1,43 @@
+package messenger;
+
+import notification.Notification;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MessengerDatabase {
+
+    private HashMap<String, String> idToNameMap;
+
+    private HashSet<Notification> storedNotifications;
+
+    public MessengerDatabase() {
```

Comment:
[kinda major but eh I dunno] Theoretically, one should not use logic as such in constructors. Firstly, if something happens, and JVM will have quit this method, it has additional work to clear after not finished/half finished object creation. Secondly it's not what user expects from a default constructor! (And I know it sounds lousy, but it is the main reason here) No one expects messing with files when calling default constructor of class.

I would strongly recommend splitting creation and reading from file.
Make constructor private and deliver some static method like `initialize`. It will create object using constructor and then fill it with values just like your constructor does now.

Reply:
So should creation be splitted like
 MessengerDatabase database = new MessengerDatabase();
database.initialize();
?

Reply:
Ah, no, ok, I understand now :D

Reply:
That way user has no way of knowing he should call initialize. Treat users as complete morons. These idiots will (I mean they really WILL) mess up in every step they can. So you have to make this unmessable to win this fight. I would so sth like this:

Inside MessengerDatabase class
```
private MessengerDatabase() {}

public static MessengerDatabase getInitialized() {
    MessengerDatabase databse = new MessengerDatabase();
    < all the rest >
}
```
Now user can only call `getInitialized` and cannot mess up. Mind that method name can be better, but I have no better idea rn.

Reply:
I said users, but this actually goes for coworkers too. Write your code in such manner that it;s usage is restrained.

Reply:
Yeah, I actually got that before you wrote it, how cool is that? :D

Reply:
Damn it. Well I had to write more stuff than you xD